src/main/java/cogni/zone/vinzclortho/CacheBodyFilter.java

```diff
@@ -0,0 +1,188 @@
+package cogni.zone.vinzclortho;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ReadListener;
+import javax.servlet.ServletException;
+import javax.servlet.ServletInputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+@RequiredArgsConstructor
+public class CacheBodyFilter implements Filter {
+  private static final Logger log = LoggerFactory.getLogger(CacheBodyFilter.class);
+
+  public static final String bodyContentProviderAttributeKey = "cogni.zone.vinzclortho.CacheBodyFilter.bodyContentProvider";
+  public static final String bodySizeProviderAttributeKey = "cogni.zone.vinzclortho.CacheBodyFilter.bodySizeProvider";
+
+  private static final byte[] emptyByteArray = new byte[0];
+  private static final int maxBytesInMemory = 1024 << 10; //1MB (I just alt-entered...)
+
+  private final RouteConfigurationService routeConfigurationService;
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws ServletException, IOException {
+    HttpServletRequest httpRequest = (HttpServletRequest) request;
+    if (routeConfigurationService.hasMatchingRoute(httpRequest)) {
+      //We patch the request because vinz is nice and lets everybody pass before him
+      //  But not everybody is as nice as vinz, and it might be that the body of the request is already consumed when we get to vinz (foei!)
+      log.debug("Vinz matches, patching request");
+      doPatchFilter(httpRequest, response, chain);
+    }
+    else {
+      log.debug("No vinz");
+      chain.doFilter(request, response);
+    }
+  }
+  private void doPatchFilter(HttpServletRequest httpRequest, ServletResponse response, FilterChain chain) throws ServletException, IOException {
+    Runnable cleanupCall = null;
```

Comment:
I have no clue what this is doing. File access and all? It looks like decompiled code btw